### ISSUE TITLE
"OS class" vs "resource profile"

### ISSUE THREAD

**Issue by fabiand** [lyarwood/common-instancetypes#1]
Great to see the scope of the common templates being replicated.
While I do wonder if we have any diff between the common templates and common instanceTypes, this epic is about how to structure the instance types.

In common-templats `server`, `desktop`, and `highperformance` were used as workload profiles.
This classification was done in order to reflect the different requirements of these workloads.
However, I wonder - I'd even argue - that server and desktop are not the /real/ workload types. The /real/ workload types are instaed a database, an AI/ML tool, an office app, whatever. And these workloads can be run on a server os just as they can be run on a desktop os.

If we agree that the instance type (and domain pref) is a key element to provide the right resources for a workload, then the instance types should not include the "os class" (server, desktop) but rather the env requirements (or resource profile): memory intensive, cpu intensive, accelerated (gpu) - and some (TBD) combinations of those.

AWS has structured their instance types similarly.

Thus examples (!) into such a direction with "resoure profiles" would be
- _**g**_eneric
- _**c**_compute (includes guaranteed cpus)
- _**m**_emory (include hugepages?)
- _**a**_ccelerated (includes some gpu?)

Which can then be used to construct the instance type names:
- g.small
- c.large
- c.tiny
- m.large
- a.xxlarge

**Comment by lyarwood** [comment]
Yup agreed, the current types and preferences are just a 0.00001 version seeking parity with the current common templates. I've made a few tweaks (like dropping `desktop` and the `tiny` `highpreformance` type) but ultimately something like what you've suggested here should be our goal.

I can hack together a PR introducing these types but I think we should continue to provide the original common templates based types and preferences for compatibility. Thoughts?

**Comment by fabiand** [comment]
And your work is great! I think it is important that we understand if instanceTypes+domPrefs can provide the same VM "template" as common templates can.

>  Thoughts?

+1 on having instances types mimicing the original common templates
At the same time I'd explore/POC the resource profile based ones.

In the end we need to decide if we want to provide the common template inspired ones as part of a release (ship them), or just in this repo (unreleased) as a fallback - and at the same time fully move over to the resource profile based ones.

My preference would be to provide the common template instancetypes for reference, but not ship them, and only ship the resource profile based ones.
Here we will need to decide if we can introduce them into kubevirt/kubevirt to make them part of stock releases (with the goal of having users to quickly adopt them).

**Comment by lyarwood** [comment]
> My preference would be to provide the common template instancetypes for reference, but not ship them, and only ship the resource profile based ones.

Why not ship them for the initial release and them deprecate them in favour of the newer approach? Once revisions are taken we don't need the resources in the cluster anymore so it's perfectly valid to deprecate and remove types over time.

> Here we will need to decide if we can introduce them into kubevirt/kubevirt to make them part of stock releases (with the goal of having users to quickly adopt them).

I wouldn't carry these types and preferences directly in `kubevirt/kubevirt` but have them installed by the virt-operator (or SSP etc) from a separate versioned repo as we currently do with `kubevirt/common-templates`.
